stackslib/src/net/download/nakamoto/download_state_machine.rs

```diff
@@ -243,18 +245,22 @@ impl NakamotoDownloadStateMachine {
         sortdb: &SortitionDB,
         loaded_so_far: &[WantedTenure],
     ) -> Result<Vec<WantedTenure>, NetError> {
-        let tip_rc = sortdb
-            .pox_constants
-            .block_height_to_reward_cycle(sortdb.first_block_height, tip.block_height)
-            .unwrap_or(0);
+        let tip_rc = downloader_block_height_to_reward_cycle(
+            &sortdb.pox_constants,
+            sortdb.first_block_height,
+            tip.block_height,
+        )
+        .expect("FATAL: tip.block_height before system start");
 
+        // careful -- need .saturating_sub(1) since this calculation puts the reward cycle start at
```

Comment:
Is this meant to be saturating_add? the line immediately following adds 1 not subs